llvm/lib/CodeGen/WasmEHPrepare.cpp

```diff
@@ -88,10 +89,12 @@
 
 using namespace llvm;
 
-#define DEBUG_TYPE "wasmehprepare"
+#define DEBUG_TYPE "wasm-eh-prepare"
 
 namespace {
-class WasmEHPrepare : public FunctionPass {
+class WasmEHPrepareImpl {
```

Comment:
I'm not familiar with the new pass manager. Does this additional `Impl` class necessary for the new pass manager? If there is a guide to port passes to the new pass manager, can you point it out? (I searched a little but didn't find much other than https://llvm.org/docs/NewPassManager.html)

Reply:
What makes the current `WasmEHPrepare` class not sharable between the legacy and the new PM and why do we need the additional `Impl` class?